### PR TITLE
Add requirements

### PR DESCRIPTION
**Pull request by huonglarne**
- Add requirements for moreh
- Modify training script

### CODE REVIEW

File: examples/scripts/sft_trainer.py [hunk 1]
```diff
@@ -99,20 +100,24 @@ class ScriptArguments:
...
 train_result = trainer.train()
+eval_result = trainer.evaluate()
 
 # Step 6: Save the model
 trainer.save_model()
 
 trainer.log_metrics("train", train_result.metrics)
 trainer.save_metrics("train", train_result.metrics)
 
+trainer.log_metrics("eval", eval_result.metrics)
```
Comment (yeomni) [inline examples/scripts/sft_trainer.py]: What kind of metrics is it expected to have? I thought we are going to use `train_loss` for key_metric

Comment (huonglarne) [reply examples/scripts/sft_trainer.py]: fixed this

File: requirements_moreh.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+-e .
```
Comment (yeomni) [inline requirements_moreh.txt:1]: it would be better if you can specify a list of libraries.

File: examples/scripts/sft_trainer.py [hunk 3]
```diff
@@ -81,7 +81,8 @@ class ScriptArguments:
 mlflow.log_param('num_params', num_params)
 
 # Step 2: Load the dataset
-dataset = load_dataset(script_args.dataset_name_or_path, split="train")
+train_dataset = load_dataset(script_args.dataset_name_or_path, split="train")
+eval_dataset = load_dataset(script_args.dataset_name_or_path, split="test")
```
Comment (yeomni) [inline examples/scripts/sft_trainer.py]: Is it necessary to have this eval dataset since we are only going to use train_loss for metric?

File: requirements_moreh.txt (hunk 2 above)
Comment (huonglarne) [reply requirements_moreh.txt:1]: when this is installed, the setup file will automatically install the repo as a package and additional libraries listed in this [file](https://github.com/moreh-dev/trl/blob/main/requirements.txt). It would be redundant if I include all the libraries in this file too

File: examples/scripts/sft_trainer.py (hunk 3 above)
Comment (huonglarne) [reply examples/scripts/sft_trainer.py]: I removed this

File: examples/scripts/sft_trainer.py [hunk 4]
```diff
@@ -99,7 +99,8 @@ class ScriptArguments:
     model=model,
     args=training_args,
     max_seq_length=script_args.seq_length,
-    train_dataset=dataset,
+    train_dataset=train_dataset,
+    eval_dataset=eval_dataset,
```
Comment (yeomni) [inline examples/scripts/sft_trainer.py]: this one should also be deleted. Please proofread your script once again as you apply changes.

Comment (huonglarne) [reply examples/scripts/sft_trainer.py]: @yeomni I deleted this and re ran the whole file to confirm it works well

**Review by yeomni — APPROVED** [review]
